cjdns-core/src/encoding.rs

```diff
@@ -31,393 +32,614 @@
 //! assert_eq!(deserialized, input);
 //! ```
 
-use std::collections::HashSet;
-
-use thiserror::Error;
-
-use crate::EncodingSchemeForm;
-
-#[derive(Error, Debug, PartialEq, Eq)]
-pub enum EncodingSchemeError {
-    #[error("Invalid encoding scheme: no encoding forms defined")]
-    NoEncodingForms,
-    #[error("Invalid encoding scheme: too many encoding forms defined (max 31)")]
-    TooManyEncodingForms,
-    #[error("Invalid encoding scheme: single form has non-empty prefix")]
-    SingleFormWithPrefix,
-    #[error("Invalid encoding scheme: form has bit_count out of bounds (1..31)")]
-    BadBitCount,
-    #[error("Invalid encoding scheme: multiple forms - prefix length is out of bounds (1..31)")]
-    MultiFormBadPrefix,
-    #[error("Invalid encoding scheme: multiple forms should have bit_count in ascending order")]
-    BitCountNotSorted,
-    #[error("Invalid encoding scheme: multiple forms must have unique prefixes")]
-    DuplicatePrefix,
-    #[error("Invalid encoding scheme: form size too big (bit_count + prefix_len > 59)")]
-    TooBigForm,
-    #[error("Invalid serialized encoding scheme")]
-    BadSerializedData,
-    #[error("Invalid encoding form")]
-    BadEncodingForm,
-}
+pub use encoding_serde::{deserialize_forms, serialize_forms};
+pub use encoding_scheme::*;
+pub use errors::{SchemeValidationError, EncodingSerDeError};
+
+mod encoding_serde {
```

Comment:
Using the word "serde" is confusing (mod name, error type name etc) because it implies using of the `serde` crate but it is not true in this case.

Reply:
I used "serde", because the contenxt of the module is `serialization` and `deserialization` of encoding schemes.
Any suggestions on names?

Reply:
still relevant after 8bd2d94

Reply:
The point is that "serde" is a name of a particular crate, and if you not using it - it is better not to confuse people.

Suggestion is just "encoding_serialization" and "EncodingSerializationError"